primitives/state-machine/src/overlayed_changes.rs

```diff
@@ -800,23 +564,28 @@ mod tests {
 
 		assert!(overlayed.storage(&key).is_none());
 
+		overlayed.start_transaction();
```

Comment:
probably does not matter, but would prefer to use something that map to existing code:
commit_prospective -> commit_tx + start_tx
rollback_prospective -> rollback_tx + start_tx
But in fact it all relate to the way we handle prospective (and in this test does not really matter).

Reply:
You cannot commit before you start a transaction. This is why it needed to be changed.

Reply:
I forgot to mention to start a transaction first, this is the same comment I did previously, you seemed to say it was doable, still it does not really matter, it was just a suggestion.